client/services/Hooks.ts

```diff
@@ -99,11 +99,27 @@ function withAuthModal(isLoggedIn: boolean, cb: (...any) => any) {
   };
 }
 
+function useKeyDown(handler: (e: KeyboardEvent) => void, eventName: string) {
+  useEffect(() => {
+    const handleEsc = (event) => {
+      if (event.code === eventName) {
+        handler(event);
+      }
+    };
+    
+    window.addEventListener('keydown', handleEsc);
+    return () => {
+      window.removeEventListener('keydown', handleEsc);
+    };
+  }, [handler]);
```

Comment:
I just realized we have this`handler` here. Since it's a function, it will make this effect run on every re-render. Here's an example: https://codesandbox.io/s/new-tree-lvfe8?file=/src/Demo.js

Let's change it to:

```
 }, [code]);
```